File: VB/DXApplication5/ISearchControlClient Implementation/SchedulerSearchProvider.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Utils
Imports DevExpress.Data.Filtering

Namespace DXApplication5
	Public Class SchedulerSearchProvider
		Inherits SearchControlProviderBase
		Protected Overrides Sub DisposeCore()
		End Sub
		Protected Overrides Function GetCriteriaInfoCore(ByVal args As SearchControlQueryParamsEventArgs) As SearchInfoBase
			Return New SchedulerSearchInfo(args.SearchText, args.FilterCondition)
		End Function
	End Class

	Public Class SchedulerSearchInfo
		Inherits SearchInfoBase
		Private text As String
		Public Sub New(ByVal t As String, ByVal condition As FilterCondition)
			text = t
			FilterCondition = condition
		End Sub
		Public Overrides ReadOnly Property SearchText() As String
			Get
				Return text
			End Get
		End Property
		Private privateFilterCondition As FilterCondition
		Public Property FilterCondition() As FilterCondition
			Get
				Return privateFilterCondition
			End Get
			Set(ByVal value As FilterCondition)
				privateFilterCondition = value
			End Set
		End Property
	End Class
End Namespace

File: CS/DXApplication5/ISearchControlClient Implementation/SchedulerSearchProvider.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Utils;
using DevExpress.Data.Filtering;

namespace DXApplication5 {
    public class SchedulerSearchProvider : SearchControlProviderBase {
        protected override void DisposeCore() {
        }
        protected override SearchInfoBase GetCriteriaInfoCore(SearchControlQueryParamsEventArgs args) {
            return new SchedulerSearchInfo(args.SearchText, args.FilterCondition);
        }
    }

    public class SchedulerSearchInfo : SearchInfoBase {
        string text;
        public SchedulerSearchInfo(string t, FilterCondition condition) {
            text = t;
            FilterCondition = condition;
        }
        public override string SearchText {
            get {
                return text;
            }
        }
        public FilterCondition FilterCondition {
            get;
            set;
        }
    }
}
